### PR TITLE
Remove unused System.Reflection.Emit leftovers

### PR DESCRIPTION
**Pull request by i3arnon**
Follow up to #56153

### CODE REVIEW

**Comment by i3arnon** [comment]
The test failure doesn't seem to be related to this change

**Comment by i3arnon** [comment]
I feel that `ScopeTree` (and possibly `LocalSymInfo`) also have no effect and can be removed, but I can't be sure.

**Comment by jeffhandley** [comment]
> The test failure doesn't seem to be related to this change

Agreed; it was an infrastructure error. I've started that check.

Tagging @dotnet/area-system-reflection-emit for a code review.

File: src/libraries/System.Reflection.Emit.ILGeneration/src/MatchingRefApiCompatBaseline.txt [hunk 1]
```diff
@@ -1,7 +1,5 @@
 Compat issues with assembly System.Reflection.Emit.ILGeneration:
 MembersMustExist : Member 'public void System.Reflection.Emit.ILGenerator.MarkSequencePoint(System.Diagnostics.SymbolStore.ISymbolDocumentWriter, System.Int32, System.Int32, System.Int32, System.Int32)' does not exist in the reference but it does exist in the implementation.
```
Comment (steveharter) [inline src/libraries/System.Reflection.Emit.ILGeneration/src/MatchingRefApiCompatBaseline.txt:2]: I assume MarkSequencePoint, GetMethodSigHelper are still necessary and must be `public` for debugging cases and\or for internal implementation. Is that correct?

**Review by steveharter — APPROVED** [review]
LGTM. Thanks for this clean-up.

Is this the final effort for this clean-up. We still have some members in the ignore.txt - I assume those are needed.

File: src/libraries/System.Reflection.Emit.ILGeneration/src/MatchingRefApiCompatBaseline.txt (hunk 1 above)
Comment (i3arnon) [reply src/libraries/System.Reflection.Emit.ILGeneration/src/MatchingRefApiCompatBaseline.txt]: `MarkSequencePoint` could be removed. `GetMethodSigHelper` overloads were used but could be made internal.

**Comment by i3arnon** [comment]
> Is this the final effort for this clean-up. We still have some members in the ignore.txt - I assume those are needed.

I'm willing to cut off as much as possible. But not sure what else is removable..

The ignore.txt errors all seem to be in the other direction (i.e. "does not exist in the implementation but it does exist in the contract"). I guess that means the API is in legacy .NET but not in core and should stay that way. Is this correct?

File: src/mono/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.Mono.cs [hunk 2]
```diff
@@ -947,20 +947,6 @@ public virtual void MarkLabel(Label loc)
...
         }
 
-        public virtual void MarkSequencePoint(ISymbolDocumentWriter document, int startLine,
-                               int startColumn, int endLine, int endColumn)
-        {
-            if (currentSequence == null || currentSequence.Document != document)
-            {
-                if (sequencePointLists == null)
-                    sequencePointLists = new List<SequencePointList>();
-                currentSequence = new SequencePointList(document);
```
Comment (lambdageek) [inline src/mono/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.Mono.cs:957]: Is the `SequencePointList` class completely unused now? Could oyu remove it, too?

(Also the commented out `GenerateDebugInfo`, below)

Comment (i3arnon) [reply src/mono/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.Mono.cs:957]: Removed

**Review by krwq — APPROVED** [review]

**Review by lambdageek — APPROVED** [review]

**Pull request merged**